### DISCUSSION TITLE
External webui inside cockpit

### DISCUSSION THREAD

**Discussion by hojnikb** [cockpit-project/cockpit#16868] [Q&A] [answered]
Would it be possible to embed an external web site (for example web ui for a specific app) inside the cockpit interface?

**Comment by martinpitt** [comment, marked as answer]
Yes, you can do that. In your page, embed the external site in an iframe, such as this:

    <iframe width="700" height="500" title="external app" src="https://yourdomain.example.com/"></iframe>

In order for browsers to actually allow that, you also need to extend the [Content-Security-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy) in your `manifest.json`:

    "content-security-policy": "default-src 'self' https://yourdomain.example.com"

I just tested that with modifying cockpit's [pinger example](https://github.com/cockpit-project/cockpit/tree/main/examples/pinger) as above:

![image](https://user-images.githubusercontent.com/200109/150759583-11f0139c-5830-4b29-b896-de42763af3af.png)
